src/main/kotlin/com/mutualmobile/praxisspringboot/controllers/orgs/impl/UserProjectApiImpl.kt

```diff
@@ -34,4 +37,38 @@ class UserProjectApiImpl : UserProjectApi {
 
         return userProjectService.assignProjectToUser(projectId = projectId, userId = userId)
     }
+
+    override fun logWorkTime(
+        projectId: String,
+        workDate: Date,
+        workHours: Int,
+        note: String?,
+        httpServletRequest: HttpServletRequest
+    ): ResponseEntity<ApiResponse<Unit>> {
+        return try {
+            val userId =
+                userDataService.getUser(httpServletRequest)?.id
+                    ?: throw Exception("No user found with the given ID!")
+
+            val userProjectId = userProjectService.findUserProject(projectId, userId)?.id
+                ?: throw Exception("Either no project exists with the given ID or the current user hasn't been assigned to it!")
+
+            val result = userProjectService.logWorkTime(
+                HarvestUserWork(
```

Comment:
@shubhamsinghmutualmobile can we instead get this object for HarvestUserWork from the API request instead of request params.